packages/editor/src/components/provider/content-only-lock-sections.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch, useRegistry } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { useEffect } from '@wordpress/element';
+import { store as blocksStore } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+/**
+ * Component that when rendered, makes it so that the editor allows only
+ * specific sections to be edited in content-only mode for templates.
+ */
+export default function ContentOnlyLockSections() {
+	const registry = useRegistry();
+
+	const { setBlockEditingMode, unsetBlockEditingMode } =
+		useDispatch( blockEditorStore );
+
+	const selected = useSelect( ( select ) => {
+		const {
+			getBlockOrder,
+			getClientIdsOfDescendants,
+			getClientIdsWithDescendants,
+		} = select( blockEditorStore );
+		const { getEditorSettings } = select( editorStore );
+		const { sectionRootClientId } = getEditorSettings();
+		const sectionClientIds = getBlockOrder( sectionRootClientId );
+		const allClientIds = sectionRootClientId
+			? getClientIdsOfDescendants( sectionRootClientId )
+			: getClientIdsWithDescendants();
```

Comment:
This code is appearing in various PRs. I think we should have a selector which finds the "sections". I don't think it's a premature abstraction because it will make the code easier to reason about.